### ISSUE TITLE
What does it mean if clustering accuracy metric fluctuates a lot?

### ISSUE THREAD

**Issue by vivektreddy** [XifengGuo/DEC-keras#19]
I am wondering what it means if the accuracy, nmi, and ari metrics fluctuate a lot.  I noticed when training on MNIST, every update interval pretty much has an improvement in accuracy and there is a upward trend.  
However, when I train on my dataset,  there are lots of fluctuations.   it sometimes starts high at iteration 0, then goes lower, then goes high again, then ends up somewhere in between.  Does this mean something is wrong with the data?  Is this trend representative of something else?

**Comment by XifengGuo** [comment]
@vivektreddy First of all, the KL loss function is not proved to be convergent. It is not surprising that the algorithm converges to a satisfactory point on some datasets but does not converge so well on others. Second, you can try to tune the `update_interval` or other hyper-parameters. If it does not work in any way, I think you should try other algorithms.
